### ISSUE TITLE
Wrong ARM thumb assembly

### ISSUE THREAD

**Issue by packmad** [Gallopsled/pwntools#1345]
```
context(os='linux', endian='little', word_size=32)

ia = asm("svc #1", arch="arm")
b'\x01\x00\x00\xef'
len(ia)
4

it = asm("svc #1", arch="thumb")
b'\x01\xdf'
len(it)
2

# so far, so good

ia = asm("mov r7, #4", arch="arm")
b'\x04p\xa0\xe3'
len(ia)
4

it = asm("mov r7, #4", arch="thumb")
b'O\xf0\x04\x07'
len(it)
4

context(os='linux', endian='little', word_size=16)
it = asm("mov r7, #4", arch="thumb")
b'O\xf0\x04\x07'
len(it)
4
```

I don't understand why it prints a 4-bytes long instruction, am I doing something wrong or is it a bug?
Edit: ah maybe it is the Thumb2 instruction? How can I get the 2-bytes instruction (I'm sure that it exists)?

**Comment by zachriggle** [comment]
This is what I get when I try the failed example, with DEBUG log level enabled.  It looks like it creates a single 4-byte instructions.  It also decodes correctly in thumb mode.

```
>>> disasm(b'O\xf0\x04\x07', mode = 'thumb')
'   0:   f04f 0704       mov.w   r7, #4'
```
```
[DEBUG] Assembling
    .section .shellcode,"awx"
    .global _start
    .global __start
    _start:
    __start:
    .syntax unified
    .arch armv7-a
    .thumb
    mov r7, #4
[DEBUG] /usr/local/bin/arm-unknown-linux-gnu-as -mthumb -EL -o /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/pwn-asm-rTumYF/step2 /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/pwn-asm-rTumYF/step1
[DEBUG] /usr/local/bin/arm-unknown-linux-gnu-objcopy -j .shellcode -Obinary /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/pwn-asm-rTumYF/step3 /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/pwn-asm-rTumYF/step4
'O\xf0\x04\x07'
```

**Comment by packmad** [comment]
But Thumb instructions should be 2-bytes long (instead Thumb2 is a mix of 2/4).

By the way, I have found a case that produces a 2-bytes long instruction:
```
asm("movs r7, #4", arch="thumb"))
b"\x04'"
```

In conclusion, I have tried to assemble `mov r7, #4` with `arm-linux-gnueabi-as` and I obtain the same 4-bytes long instruction, so it is not a problem of pwntools.

I close the issue in the meantime, but if some of the readers can explain to me what's wrong in my understanding of ARM/Thumb... will be appreciated!
